src/ser/mod.rs

```diff
@@ -61,49 +61,65 @@ impl<'a, W: Write, O: Options> serde::Serializer for &'a mut Serializer<W, O> {
     }
 
     fn serialize_u16(self, v: u16) -> Result<()> {
-        self.writer.write_u16::<O::Endian>(v).map_err(Into::into)
+        self.writer
+            .write_u16::<<O::Endian as BincodeByteOrder>::Endian>(v)
```

Comment:
If you wanted to you could be fancy and create newtype
`type GetEndian<O> = <<O as Options>::Endian as BincodeByteOrder>::Endian;`
Then replace all these calls with
`.write_u16::<GetEndian<O>>(v)`

Reply:
I'll leave this alone for now.